File: src/tidy_claude/state.py
"""Mutable run-time state passed through the CLI."""

import click


class RunState:
    def __init__(self, debug: bool = False):
        self.debug = debug
        self.stats: dict[str, int] = {}

    def log(self, msg: str):
        if self.debug:
            click.echo(msg)

    def count(self, key: str, n: int = 1):
        self.stats[key] = self.stats.get(key, 0) + n

File: src/state.rs
// state.rs — RunState mutable runtime state (to be implemented in Phase 2)

#[derive(Default)]
pub struct RunState {
    pub debug: bool,
    pub stats: std::collections::HashMap<String, usize>,
}

impl RunState {
    pub fn log(&self, msg: &str) {
        if self.debug {
            eprintln!("[debug] {msg}");
        }
    }

    pub fn count(&mut self, key: &str, n: usize) {
        *self.stats.entry(key.to_string()).or_default() += n;
    }
}
